lib/creditsafe/client.rb

```diff
@@ -86,6 +91,15 @@ def invoke_soap(message_type, message)
       notification_payload[:response] = response.body
     rescue => raw_error
       processed_error = handle_error(raw_error)
+
+      if processed_error.class == AccountError
+        retries -= 1
+        if retries.positive?
+          sleep(RETRY_SLEEP_TIME)
```

Comment:
I don't think this is the right approach, especially since our history of seeing these exceptions suggests that the credential failures come in bursts. We shouldn't block execution while waiting for the transient error to pass, we should return the work to our work queue via the retry system and take advantage of its exponential-backoff capabilities.  Non-GoCardless consumers of this library also should not suddenly see the code going to `sleep` for up to 15 seconds by default in the event of a transient Creditsafe error or a misconfiguration, as this represents a substantial change in the library's behaviour.